cibuildwheel/options.py

```diff
@@ -584,8 +592,10 @@ def config_file_path(self) -> Path | None:
 
     @functools.cached_property
     def package_requires_python_str(self) -> str | None:
-        args = self.command_line_arguments
-        return get_requires_python_str(Path(args.package_dir))
+        return get_requires_python_str(self.project_dir, self.pyproject_toml)
+
+    def dependency_groups(self, *groups: str) -> tuple[str, ...]:
+        return get_dependency_groups(self.pyproject_toml, *groups)
```

Comment:
Slightly pedantic (apologies), but these method names confused me, I thought they were returning some kind of group structure but really they're returning package names. Could `resolve_dependency_groups` be a better name for `get_dependency_groups`? Also, I'm not sure this method is needed, the `projectfiles` function could be called directly.